src/prefect/context.py

```diff
@@ -64,6 +70,34 @@
 # default settings.
 GLOBAL_SETTINGS_CONTEXT = None  # type: ignore
 
+logger = get_logger("context")
+
+
+def log_non_production_warning(ctx: "ClientContext"):
```

Comment:
note for future me: if we try this again it might do better in the `app.lifespan` as opposed to the `clientcontext`